### PR TITLE
Fix iOS last up event velocity calculation - version 2

### PR DESCRIPTION
**Pull request by xster**
Alternate to https://github.com/flutter/flutter/pull/11345 for fix of https://github.com/flutter/flutter/issues/10977 by checking if last moved -> ended settling is within threshold. 

Will add tests after a first pass.

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm [hunk 1]
```diff
@@ -407,6 +407,19 @@ - (void)dispatchTouches:(NSSet*)touches phase:(UITouchPhase)phase {
...
+    // The touch ended location can settle since the last touch moved location but without the
+    // precise timestamp of when the location settled which will throw off velocity calculation
+    // in Flutter. If the settling distance is within the margin of error from the previous
+    // location, just report the previous location on touch ended.
+    if (touch.phase == UITouchPhaseEnded) {
+      CGPoint endedLocation = [touch locationInView:nil];
+      CGPoint lastMovedLocation = [touch previousLocationInView:nil];
+      CGFloat distanceSquared = pow(endedLocation.x - lastMovedLocation.x, 2)
+          + pow(endedLocation.y - lastMovedLocation.y, 2);
+      if (distanceSquared < pow(touch.majorRadiusTolerance, 2))
```
Comment (Hixie) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: is `pow(x, 2)` as fast as `x * x` in ObjectiveC?

**Comment by Hixie** [comment]
...test? :-)

File: shell/platform/darwin/ios/framework/Source/FlutterViewController.mm (hunk 1 above)
Comment (xster) [inline shell/platform/darwin/ios/framework/Source/FlutterViewController.mm]: It's just the C pow. https://stackoverflow.com/questions/6321170/is-there-any-advantage-to-using-powx-2-instead-of-xx-with-x-double seems split. Switched to `x * x` to be safe.

**Comment by Hixie** [comment]
I gave him more or less the opposite feedback on the framework side. :-)

My understanding is that it's iOS-specific, in that on iOS the data is not as accurate as on Android. (We don't get this problem with the Android data.) But I don't have a full understanding of the issue.

**Comment by xster** [comment]
Closing. We're going for ##11571

**Pull request closed without merging**